014-realtime-api.md

```diff
@@ -90,8 +91,67 @@ Possible protocols to take under future considiration:
+| collections.[ID].documents | Any update/delete/create events to any document in a given collection where user has read access |
+| documents                  | Any update/delete/create events to documents where user has read access |
+| documents.[ID]             | Any update/delete/create events to a given document where user has read access |
+| files                      | Any update/delete/create events to file where user has read access |
+| files.[ID]                 | Any update/delete/create events to a given file where user has read access |
+| functions                  | Any execution for functions where user has read access       |
+| functions.[ID]             | Any execution event to a given function where user has read access |
+| functions.executions       | Any update to executions where user has read access          |
+| functions.executions.[ID]  | Any update to a given function execution where user has read access |
+
+### Messages
+
+The message from the server to the clients should be a JSON string and reflect the following object:
+
+| Property  | Description                                                  |
+| --------- | ------------------------------------------------------------ |
+| event     | The name of the event equivalent to the subscribed to channel name. |
+| timestamp | To ensure consistency across all client platforms and real-time technologies, the event timestamp is included. |
+| payload   | Payload contains the data equal to the response model.       |
+
+### Subscriptions
+
+#### Websocket
+
+Subscriptions can be bound to channels and are passed as query parameters and part of the websocket URL. An example looks like this:
+
+`wss://appwrite.test/v1/realtime?project=XXXXXX&channels[]=account&channels=documents.XXXXXX`
+
+Or as SDK call in JavaScript:
 
+```js
+let sdk = new Appwrite();
+
+sdk
+    .setHostname('appwrite.test')
+    .setProject('5df5acd0d48c2') // Your project ID
+    .setVersion('v1') // Optional and v1 by default
+    .setSecureProtocol(true) // Optional and true by default - will use https/wss or http/ws
+;
+
+sdk.realtime.subscribe('account', response => {
+  console.log(response); // Callback will be executed on account event.
+});
+
+const documentsUnsubscribe = sdk.realtime.subscribe('documents.XXXXXX', response => {
+  console.log(response); // Callback will be executed on documents.XXXXXX event
+});
+
+documentsUnsubscribe(); // the subscribe() method will return a method to unsubscribe, invalidate the callback and remove the channel
+```
+
+As seen, the initialization of the client SDK changes a lot. Instead of specifying the WebSocket URI, we can work much more flexibly via the hostname and customize the to be used protocols in the future without another breaking change on the client SDKs.
+
+The client SDK will maintain a single connection and will maintain all the channels, if a channel is added or removed - a new connection will be established with given channels.
```

Comment:
Yep, I like this approach. Reduces complexity by a lot, at least for the first phase.